### ISSUE TITLE
Consider a name change

### ISSUE THREAD

**Issue by ericwooley** [SpencerCDixon/redux-cli#46]
As big of a fan of redux as I am, this project got more general with the introduction of blueprints. I think blueprints got a lot of things right that something like YO got wrong, specifically committing the blueprints into the project. So I could see this project taking off. 

I intend to use this on several projects which are not redux based. (I work in marketing and have to deal with dozens of projects with all kinds of setups).

This is still a really young project, it might be a good idea to change the name while you still can? It's going to be confusing if this is still going in 5 years and no one is using redux.

**Comment by SpencerCDixon** [comment]
Yeah I 100% agree.  When I reached out to some people to see if they wanted to contribute first thing they said was they didnt want to work on a project with a hard dep on Redux but this has no dep on redux haha.  Have any suggestions for names??

**Comment by ericwooley** [comment]
Hmm, this is always a hard question.

Well given that it's pretty blueprint centric now, maybe something related to architecture and construction? Maybe mason-cli? That might be too close to maven though. Arc (architect)? I don't know.

You could also try to jump back on the re-* train. 'reyo' for example. 

Not sure, I'll keep thinking about it.

**Comment by SpencerCDixon** [comment]
All thought provoking questions :+1: I'll keep thinking about it as well!  It would be nice to tie it directly to the React ecosystem if possible.  If I had a few others willing to spend a good amount of time I'd love for the CLI to build blueprints that were very opinonated to React development in order to start going more in the direction of ember-cli.  

At that point though it might have just made more sense to extend or fork ember-cli.  I know they are putting considerable effort into making it just a generic cli-generator type of thing and angular has already forked it and started making it their own.  

Bah! decisions, decisions.. haha

**Comment by ericwooley** [comment]
It would be nice either way, as long as the templates are committed and customizable. I would be a happy camper. It's been something that I have wanted for quite a while. 

If ember created a generic one that was polished, it might be a good idea incorporate that base.

**Comment by ericwooley** [comment]
So a couple I keep coming back too.

1. Repro-cli
2. Regen-cli
3. ReArc-cli

personally I like having the re in the names. but maybe you would rather not. 

Anyways, thats my input!

**Comment by davezuko** [comment]
Also ref: https://github.com/SpencerCDixon/redux-cli/issues/29

**Comment by anakinjay** [comment]
Big fan of Regen-cli  React Generator has a nice ring to it.

**Comment by ericwooley** [comment]
Maybe we can make use of these reactions as a way to vote. Upvote and downvote your favorites and unfavorites respectively.

**Comment by ericwooley** [comment]
Repro-cli

**Comment by ericwooley** [comment]
Regen-cli

**Comment by ericwooley** [comment]
ReArc-cli

**Comment by SpencerCDixon** [comment]
Yeah I'm a fan of regen-cli as well.  Reached out to the repo owner of react-cli to see if he would be willing to figure out a way for us to use that.  We'll see what happens!

**Comment by gabeweaver** [comment]
Just throwing in my $.02. I just started using redux-cli tonight but already see the immense value and am honestly sad it took me as long as it has to stumble across the progress you're making. I see the value in tying the name to React, but I have another suggestion:

`paradigm-cli`

Why? 

1. The philosophy behind React revolves around the component. The component is this re-usable thing that is limited only by the creativity and skill of the engineer building it. 
2. [Paradigm Defined](http://www.dictionary.com/browse/paradigm) 
3. --Para is an [appropriate prefix](http://www.dictionary.com/browse/para-) for a CLI. It's never the star of the show, but an essential auxiliary. 

Thanks for putting the project out here. I'll definitely want to contribute as soon as my learning curve tapers off.

**Comment by mzvast** [comment]
Is this a naming brainstorm?
So it's my turn aha~
 `Red-cli` is a good one.For 3 reasons:
1.It's short to type and easy to remember as well as no conflict to current name.
2.Red is a color, symbol for hope?
3.There is a famous battle named 'Red-cliff' 

:)

**Comment by ArnaudValensi** [comment]
This repo is awesome, thanks!

Like some other people, I started using this repo for different js projects, even not redux or react. The blueprint system is very simple, we can commit our own blueprints and everybody can get how it works in no time.
I really believe that it would be awesome to make this tool very generic. And on the other hand, it is very cool to have something very tied to the react/redux ecosystem.

A very cool thing would be making two layers:
1. the generic generator in one side
2. the redux-cli on the top

This way, anybody could use the generic generator to any kind of project, and why not create different cli for different architectures/boilerplates.

What do you think?

I can help, I can pull a branch if you are up to. :)

**Comment by SpencerCDixon** [comment]
Yeah I'm definitely down.  Extract the generator functionality into a separate name and then just create a layer on top of it that is specific to react/redux related generators.  I should start to have some more free time now to be able to work on something like that.. lets coordinate @ArnaudValensi

**Comment by jamesr73** [comment]
Really like where this is going. I'm just starting out with React/Redux and I think this will be a great help (as will the starter kit!). I have a bit more experience with CLIs so I'm hoping I can find a way to help out here. My suggestion:

rebl

Thoughts:
1. keeps nod to re-act/dux
2. bl hints at blueprint
3. rebl hints at blueprints to create blueprints (super meta! ;-)
4. rebl is short and easy to type (better than e.g. rebp)

Cons:
1. too close to rebel code generator?

Aside:
If a generic blueprint generator is separated from the redux/react blueprints do they become a layer on top or a plugin/companion? I'm thinking companion modules could provide additional/specific blueprints and/or new/init targets and the core has the ability to use these directly from the module or pull them into the project for customisation. Projects can then mix-n-match blueprints for whatever frameworks/languages they choose to use.

**Comment by jonricaurte** [comment]
I think the name should be **blueprint-cli**. Simple and explains the features being provided.

**Comment by ArnaudValensi** [comment]
I like **blueprint-cli**.

**Comment by xml** [comment]
In the field of architecture, the young employee who works on blueprints for the skilled architect is called a 'draftsman'. A Google search for "npm draftsman" yields no results...

**Comment by thejase** [comment]
*colony-cli*

The usage and commands generally fit the purpose of insect colonies:
```
npm install -g colony-cli
colony init
cd app
colony generate dumb Button
```

**Comment by anithri** [comment]
continuing from #24 

As we start the dev cycle for 2.0, it's time to rename this.  I think **blueprint-cli** is the safest bet.  There are other blueprinty-names on npm, but none of them look like commands, unlike the type of projects that **regen** was showing.  Also, **bp** as an alias is mighty enticing.

**Comment by SpencerCDixon** [comment]
Yeah I'm cool to give the green light on *blueprint-cli* with the alias as bp

**Comment by anithri** [comment]
addressed by #109
